packages/core/src/Users.ts

```diff
@@ -1,137 +1,406 @@
+      localStorage.removeItem('youversion-auth-state');
+      throw error;
+    }
+  }
+
+  /**
+   * Redirects to the server callback endpoint to obtain authorization code
+   */
+  private static obtainLocation(callbackURL: string, state: string): void {
+    const url = new URL(callbackURL);
+    const params = new URLSearchParams(url.search);
+
+    if (params.get('state') !== state) {
+      throw new Error('Invalid state parameter');
+    }
+
+    // Redirect to the server callback endpoint with all the current parameters
+    const serverCallbackUrl = new URL(
+      `https://${YouVersionPlatformConfiguration.apiHost}/auth/callback`,
+    );
+    params.forEach((value, key) => {
+      serverCallbackUrl.searchParams.set(key, value);
+    });
+
+    window.location.href = serverCallbackUrl.toString();
+  }
+
+  /**
+   * Extracts sign-in result from token response
+   */
+  private static extractSignInResult(tokens: {
+    access_token: string;
+    expires_in: number;
+    id_token: string;
+    refresh_token: string;
+    scope: string;
+    token_type: string;
+  }): SignInWithYouVersionResult {
+    const idClaims = this.decodeJWT(tokens.id_token);
+
+    const permissions = tokens.scope
+      .split(' ')
+      .map((p) => p.trim())
+      .filter((p) => p.length > 0)
+      .filter((p): p is SignInWithYouVersionPermissionValues =>
+        Object.values(SignInWithYouVersionPermission).includes(
+          p as SignInWithYouVersionPermissionValues,
+        ),
+      );
+
+    const resultData = {
+      accessToken: tokens.access_token,
+      expiresIn: tokens.expires_in,
+      refreshToken: tokens.refresh_token,
+      idToken: tokens.id_token,
+      permissions,
+      yvpUserId: idClaims.sub as string,
+      name: idClaims.name as string,
+      profilePicture: idClaims.profile_picture as string,
+      email: idClaims.email as string,
+    };
+
+    return new SignInWithYouVersionResult(resultData);
+  }
+
+  /**
+   * Decodes JWT token payload
+   */
 
-    // Use the registered authentication strategy
-    const strategy = AuthenticationStrategyRegistry.get();
-    const callbackUrl = await strategy.authenticate(url);
-    const result = new SignInWithYouVersionResult(callbackUrl);
+  private static decodeJWT(token: string): Record<string, any> {
+    const segments = token.split('.');
 
-    if (result.accessToken) {
-      YouVersionPlatformConfiguration.setAccessToken(result.accessToken);
+    if (segments.length !== 3) {
+      return {};
     }
 
-    return result;
+    let base64 = segments[1]?.replace(/-/g, '+').replace(/_/g, '/');
+
+    while (base64 && base64.length % 4 !== 0) {
+      base64 += '=';
+    }
+
+    try {
+      if (base64) {
+        const data = atob(base64);
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return JSON.parse(data);
+      } else {
+        return {};
+      }
+    } catch {
+      return {};
```

Comment:
In dev mode, should we log JWT errors instead of failing silently?

```suggestion
      if (process.env.NODE_ENV === 'development') {
        console.error('JWT decode failed:', error);
      }
      return {};
```